Container factory method

darlison-alves opened the pull request:
Container class created to manage "Handler class" and dependency injection in the same.
How exemple, I writed some tests....

darlison-alves committed:
interfaces..

darlison-alves committed:
implementing container logic...

darlison-alves committed:
fix....

Commit:
fix...

darlison-alves committed:
escrevendo testes para container ...

darlison-alves committed:
implementando classe Container na Classe Mercury...

darlison-alves committed:
adaptando messageBus ao container...

darlison-alves committed:
lançando exeções do container...

darlison-alves committed:
fix...

darlison-alves committed:
2.0.14

darlison-alves committed:
fix...

darlison-alves committed:
Update README.md

darlison-alves committed:
fix...

darlison-alves committed:
Merge branch 'container-factory-method' of https://github.com/Monsterholic/mercury into container-factory-method

cl3dson commented:
Many new issues in codify code quality review, please could you check some of them?

Commit:
fix...

iagosrodrigues commented:
You could just see in the details [here](https://app.codacy.com/manual/f.cl3dson/mercury/pullRequest?prid=5070860) and fix the issues.

Before the merge, please squash them up or use a fixup. This is required to remove the "fix" commits that don't have helpful messages.

iagosrodrigues reviewed (changes requested):
Have to fix the Codacy issues before merge.

Commit:
fix...

fix...

fix..

fix..

fix..

fix..

fix...

Dfix..

fix..

fix..

fix..

fix..

fix..

fix...

fix..

fix..

fix...

fix..

Commit:
2.1.0

iagosrodrigues commented:
I'll see this at the weekend.

darlison-alves committed:
correcing written tests

iagosrodrigues commented on lib/connection/RabbitMQConnectionFacade.ts:
```diff
@@ -153,14 +157,17 @@ export class RabbitMQConnectionFacade {
 
     public async dispatchMessage(descriptor: string, msg: ConsumeMessage): Promise<void> {
-        const handlers = Mercury.handlerRegistry;
+        //const handlers = Mercury.handlerRegistry;
+
```
I think you should remove the commented code as well.
```suggestion
```

iagosrodrigues commented on lib/connection/RabbitMQConnectionFacade.ts:
```diff
@@ -153,14 +157,17 @@ export class RabbitMQConnectionFacade {
             if (handlerClassName) {
-                if (handlers.has(handlerClassName)) {
-                    const handler = handlers.get(handlerClassName);
+                //if (handlers.has(handlerClassName))
+                if (handler) {
+                    //const handler = handlers.get(handlerClassName);
```
Here too.
```suggestion
                if (handler) {
```

iagosrodrigues reviewed (changes requested):
Here more tips.
